### PR TITLE
Replace TYPO3_DB-Calls with Doctrine and add functional testing

### PR DESCRIPTION
**Pull request by fseipel**

### CODE REVIEW

**Comment by mbrodala** [comment]
Looks good for me but I'd also like @pm-sschreiber to have a look.

File: Classes/Domain/Repository/FileReferenceRepository.php [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?php
```
Comment (pm-sschreiber) [inline Classes/Domain/Repository/FileReferenceRepository.php]: Why is this repository needed? Cannot find it inside the code changes.

File: Classes/Domain/Repository/LanguageSkillRepository.php [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<?php
```
Comment (pm-sschreiber) [inline Classes/Domain/Repository/LanguageSkillRepository.php]: Same here - why this repository?

File: Classes/TCA/FormHelper.php [hunk 3]
```diff
@@ -62,18 +63,20 @@ public function findContributors(&$params)
...
-        ]);
+        $queryBuilder = $this->getQueryBuilder('be_groups');
+        $queryBuilder->select('uid', 'title')
+            ->from('be_groups')
+            ->where($queryBuilder->expr()->eq('tx_ats_location', $queryBuilder->createNamedParameter($params['row']['location'])));
 
         if (ExtensionManagementUtility::isLoaded('extbase_acl') && !empty($GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['ats']['job']['roles']['department'])) {
-            $where .= ' AND be_groups.tx_extbaseacl_role IN(' . $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['ats']['job']['roles']['department'] . ')';
+            $queryBuilder->andWhere(
+                $queryBuilder->expr()->in('tx_extbaseacl_role', $queryBuilder->createNamedParameter($GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['ats']['job']['roles']['department']))
```
Comment (pm-sschreiber) [inline Classes/TCA/FormHelper.php]: Please use ExtconfService->getInstance()->getJobRoleDefinitions()[...] instead (https://github.com/pagemachine/ats/blob/master/Classes/Service/ExtconfService.php#L45)

File: Classes/TCA/FormHelper.php [hunk 4]
```diff
@@ -93,62 +96,56 @@ protected function findUserByGroupRoleAndLocation($role, $location)
...
-     * Helper function to mock static BackendUtility function
+     * Helper function to mock QueryBuilder
      * @codeCoverageIgnore
      *
      * @param  string $table
-     * @return string
+     * @return object
      */
-    public function getBackendEnableFields($table)
+    public function getQueryBuilder($table)
```
Comment (pm-sschreiber) [inline Classes/TCA/FormHelper.php]: Do we really need this helper function? You could instance ConnectionPool once and use it directly.

File: Classes/Domain/Repository/FileReferenceRepository.php (hunk 1 above)
Comment (pm-sschreiber) [reply Classes/Domain/Repository/FileReferenceRepository.php]: Will be resolved later once I find a solution.

**Pull request merged**